### ISSUE TITLE
Blacklist service from serviceset for host template fails

### ISSUE THREAD

**Issue by bengelberth** [Icinga/icingaweb2-module-director#1707]
## Expected Behavior

When blacklisting a Service from a Service Set at the Host Template level, the service should not be applied to hosts that import the host template.  It should also be striked through when looking at the Services on the Host in Director.

## Current Behavior

The blacklisted Service is still applied to the host.  The Service is still listed when editing the Host in director.

## Possible Solution

The service is not being ignored on the host because Director is producing code that is "ignore where host.name == "host template name" rather than ignore where "host template name" in host.templates within zones.d/director-global/servicesets.conf

IcingaService.php renderCustomExtensions() is not checking if it is a template or host object before inserting the ignore.

I have not looked around the code to see how to correct the strike through issue when looking at the Services on the Host in Director.

## Steps to Reproduce (for bugs)

Create a Service Set
Add the Service Set to a Host Template
Black list a Service from the Service Set in the Host Template
Import the Host Template from a Host

The Service will still be active.

## Your Environment

* Director version (System - About): 1.5.2
* Icinga Web 2 version and modules (System - About): 2.6.1, Cube 1.0.1, fileshipper 1.1.0
* Icinga 2 version (`icinga2 --version`): 2.10.2
* Operating System and version: CentOS 7
* Webserver, PHP versions: httpd-2.4.6-80.el7.centos.1.x86_64, rh-php71-php-fpm-7.1.8-1.el7.x86_64

**Comment by jwilliams-shadowsoft** [comment]
This issue seems to occur when mapping a Service Set to a Host as well.

## Steps to reproduce
1. Map a Service Set to a specific Host:
`Icinga Director => Host objects => Host => (specific Host) => Services => Add service set`
2. Then, blacklist one Service from the Service Set for that Host:
`Icinga Director => Hosts => (specific host) => Services => (specific service) => Blacklist`

## Failure results
* An `ignore where` statement to exclude the Host is inserted into the `object Service` block for the Service-to-Host mapping. This causes a syntax check failure upon deployment since there is no `apply where` statement in an `object Service` definition.
* The Service is shown (incorrectly) in strikeout in the web interface (apparently because it was marked as "blacklisted" in the database).
* The Service continues to be monitored in the usual way, even though it was supposedly blacklisted.
* The configuration is invalid because the deployment failed.

## What works
1. Map a Service Set to a Host using an `assign where` rule.
`Icinga Director => Monitored Services => Service Sets => (specific service set) => Assign where`
2. Then blacklist one Service in the same way:
`Icinga Director => Hosts => (specific host) => Services => (specific service) => Blacklist`

## Success results
* An `ignore where` statement` to exclude the Host is inserted into the `assign Service` block for the Service-to-Host mapping. This works because the block can accept `ignore where` and already has `apply where` statements.
* The Service is shown (correctly) in strikeout in the web interface.
* The Service is no longer in the monitoring config and does not appear in the monitoring interface.
* The configuration is valid and the deployment succeeds.

This was tested with Director 1.6.0 on an Icinga 2.10.2 platform.

**Comment by Thomas-Gelf** [comment]
This is now possible:

* **Base Host Template**: assigned a `Service Set` with the **Dummy 2** `Service`
* **Host Template**: inherit from **Base Host Template**, blacklist **Dummy 2**
* **some.host.example.com**: inherit from **Another Template**, blacklist **Dummy 2**

Renders as follows:
```
apply Service "Dummy 2" {
    import "Dummy"

    assign where "Base Host Template" in host.templates
    ignore where host.name == "some.host.example.com"
    ignore where "Host Template" in host.templates

    import DirectorOverrideTemplate
}
```

Will be pushed immediately, merged into support/1.6 and released with v1.6.1.
